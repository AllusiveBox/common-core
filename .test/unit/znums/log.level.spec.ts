import "jest-extended";
import { LogLevel } from "../../../.src";

/**
 *
 * Test Suite for the LogLevel class.
 *
 * @group unit
 * @group znum
 */
describe(
    "LogLevel Unit Test Suite",
    () => {

        test(
            'that LogLevel static toString returns "LogLevel"',
            () => {
                expect(LogLevel.toString())
                    .toBe("LogLevel");
            }
        );

        test(
            "that SUPPORTED_ZNUMS is the correct shape and size",
            () => {
                const values: Array<LogLevel> = LogLevel.SUPPORTED_ZNUMS;

                expect(values)
                    .toBeArrayOfSize(6);
                values.forEach((value: LogLevel): void => {
                    expect({
                        code: value.code,
                        logLevel: value.logValue
                    }).toMatchSnapshot();
                });
            }
        );

        test.each([
            "fatal",
            "error",
            "warn",
            "info",
            "verbose",
            "debug"
        ])
        ("that given %s, getLogLevel returns the expected result", (arg) => {
            expect(LogLevel.getLogLevel(arg))
                .toMatchSnapshot();
        });

    }
);