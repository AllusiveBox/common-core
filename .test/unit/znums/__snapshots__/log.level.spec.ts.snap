// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogLevel Unit Test Suite that SUPPORTED_ZNUMS is the correct shape and size 1`] = `
{
  "code": "FATAL",
  "logLevel": 0,
}
`;

exports[`LogLevel Unit Test Suite that SUPPORTED_ZNUMS is the correct shape and size 2`] = `
{
  "code": "ERROR",
  "logLevel": 1,
}
`;

exports[`LogLevel Unit Test Suite that SUPPORTED_ZNUMS is the correct shape and size 3`] = `
{
  "code": "WARN",
  "logLevel": 2,
}
`;

exports[`LogLevel Unit Test Suite that SUPPORTED_ZNUMS is the correct shape and size 4`] = `
{
  "code": "INFO",
  "logLevel": 3,
}
`;

exports[`LogLevel Unit Test Suite that SUPPORTED_ZNUMS is the correct shape and size 5`] = `
{
  "code": "VERBOSE",
  "logLevel": 4,
}
`;

exports[`LogLevel Unit Test Suite that SUPPORTED_ZNUMS is the correct shape and size 6`] = `
{
  "code": "DEBUG",
  "logLevel": 5,
}
`;

exports[`LogLevel Unit Test Suite that given debug, getLogLevel returns the expected result 1`] = `LogLevel {}`;

exports[`LogLevel Unit Test Suite that given error, getLogLevel returns the expected result 1`] = `LogLevel {}`;

exports[`LogLevel Unit Test Suite that given fatal, getLogLevel returns the expected result 1`] = `LogLevel {}`;

exports[`LogLevel Unit Test Suite that given info, getLogLevel returns the expected result 1`] = `LogLevel {}`;

exports[`LogLevel Unit Test Suite that given verbose, getLogLevel returns the expected result 1`] = `LogLevel {}`;

exports[`LogLevel Unit Test Suite that given warn, getLogLevel returns the expected result 1`] = `LogLevel {}`;
